examples/cloudml-bee-health-detection/trainer/model.py

```diff
@@ -65,19 +65,16 @@ def create_classifier(config, parameters):
     Returns:
         `tf.estimator.DNNClassifier` with specified features and architecture.
     """
-    # Mean and Standard Deviation Constants for normalization.
-    depth_mean = np.float32(parameters.depth_mean)
-    depth_std = np.float32(parameters.depth_std)
-
     # Columns to be used as features.
-
-    depth = tf.feature_column.numeric_column(
-        'depth',
-        normalizer_fn=(lambda x: (x - depth_mean) / depth_std))
+    subspecies = tf.feature_column.categorical_column_with_vocabulary_list(
```

Comment:
Mmm.  Maybe similar to how 'mean' and 'std-dev' had to be passed to the salt model, this feels like the wrong place to define a vocabulary for subspecies.  Similarly, is there not much we can do here without a massive amount of overengineering?

Reply:
I'll add these as constants at the top so it is easier to change.

This is not similar to mean/std as newer data (fro the same problem) shouldn't change these values.

mean and std will always change for new data (even for the same problem)

Reply:
If newer training data contains a bigger subspecies vocabulary, does it change these values here?

In any event, I like the indirection you've introduced.

I don't think my concern requires any changes.  But similar to as with mean/std, I'm made vaguely uncomfortable by the model having to know not only the shape of the training data but also its contents.  I don't see any way straightforward way around this, and so this is probably just a good lunchtime conversation.

Reply:
I added code to use the "Other" embedding if the value is not in the vocabulary.
This way it won't break if a new value shows up.